docs/website/docs/plus/getting-started/tutorial.md

```diff
@@ -0,0 +1,395 @@
+To count rows in the dataset, run:
+
+```sh
+dlt dataset my_pipeline_dataset row-counts
+```
+
+This will show the number of rows in the items table as specified by the arrow source. Additionally, the internal dlt tables are shown.
+
+```sh
+            table_name  row_count
+0                items        100
+1         _dlt_version          1
+2           _dlt_loads          1
+3  _dlt_pipeline_state          1
+```
+
+### View data
+To view the first five rows of the `items` table:
+
+```sh
+dlt dataset my_pipeline_dataset head items
+```
+
+This displays the top entries in the `items` table, enabling quick validation of the pipeline's output. The output will be something like this:
+
+```sh
+Loading first 5 rows of table items.
+
+   id   name  age
+0   0  jerry   49
+1   1    jim   25
+2   2   jane   46
+3   3   john   48
+4   4  jenny   49
+```
+
+To show more rows, use the `--limit` flag.
+
+```sh
+dlt dataset duckdb_dataset head items --limit 50
+```
+
+## Adding sources, destinations, and pipelines to your project
+
+Adding a new entity to an existing dlt+ Project is easy. You can add a new entity to your project by running the `dlt <entity_type> <entity_name> add` command. Depending on the entity you are adding, different options are available. To see all options for adding a destination, for example, you can run `dlt destination add --help`. Let's individually add a source, destination, and pipeline to a new project, replicating the default project we created in the previous chapter.
+
+### Create an empty project
+
+Delete all the files in the `tutorial` folder and run the following command to create an empty project:
+
+```sh
+dlt project init
+```
+
+This will create a project without any sources, destinations, datasets, or pipelines; the project will be named after the folder.
+
+### Add all entities
+
+Now we can add all of our entities individually. This way, we can also give them their own names, which will be useful when having multiple destinations of the same type, for example.
+
+Add a source:
+
+```sh
+# add a new arrow source called "my_arrow_source"
+dlt source my_arrow_source add arrow
+```
+
+Add a destination:
+
+```sh
+# add a new duckdb destination called "my_duckdb_destination"
+# this will also create a new dataset called "my_duckdb_destination_dataset"
+dlt destination my_duckdb_destination add duckdb
+```
+
+If you want dlt+ also to add a definition of dataset for this specific destination, you can use the `--dataset-name` flag:
+
+```sh
+# add a new duckdb destination called "my_duckdb_destination"
+dlt destination my_duckdb_destination add duckdb --dataset-name my_duckdb_destination_dataset
+```
+
+In this case both destinations and datasets sections of the `dlt.yml` file will be updated:
+
+```yaml
+# your destinations are the databases where your data will be saved
+destinations:
+  my_duckdb_destination:
+    type: duckdb
+
+# your datasets are the datasets on your destinations where your data will go
```

Comment:
I'm not sure how to solve this, but mentioning how to create datasets might be confusing if we do not really explain how they work or why one would use them.

Reply:
Maybe a note that dataset docs are coming..